msteams-platform/toolkit/configure-message-extension-capability.md

```diff
@@ -0,0 +1,634 @@
+                   "type": "action",
+                   "parameters": [
+                       {
+                           "name": "title",
+                           "title": "Card title",
+                           "description": "Title for the card",
+                           "inputType": "text"
+                       },
+                       {
+                           "name": "subTitle",
+                           "title": "Subtitle",
+                           "description": "Subtitle for the card",
+                           "inputType": "text"
+                       },
+                      {
+                           "name": "text",
+                           "title": "Text",
+                           "description": "Text for the card",
+                           "inputType": "textarea"
+                       }
+                   ]
+               },
+               {
+                   "id": "shareMessage",
+                   "context": [
+                       "message"
+                   ],
+                   "description": "Test command to run action on message context (message sharing)",
+                   "title": "Share Message",
+                   "type": "action",
+                   "parameters": [
+                       {
+                           "name": "includeImage",
+                           "title": "Include Image",
+                           "description": "Include image in Hero Card",
+                           "inputType": "toggle"
+                       }
+                   ]
+               },
+               {
+                   "id": "searchQuery",
+                   "context": [
+                       "compose",
+                       "commandBox"
+                   ],
+                   "description": "Test command to run query",
+                   "title": "Search",
+                   "type": "query",
+                   "parameters": [
+                       {
+                           "name": "searchQuery",
+                           "title": "Search Query",
+                           "description": "Your search query",
+                           "inputType": "text"
+                       }
+                   ]
+               }
+           ],
+           "messageHandlers": [
+               {
+                   "type": "link",
+                   "value": {
+                       "domains": [
+                           "*.botframework.com"
+                       ]
+                  }
+               }
+           ]
+       }
+   ]
+   ```
+
+## Bring message extension code to project
+
+If you're adding message extension to a bot Teams app, you should have a class that extends `TeamsActivityHandler`. Bring your message extension functions or copy functions from your previously created message extension app to your own class. The following code is an example if you copy functions from a message extension app created with Teams Toolkit:
```

Comment:
Copy your message extension functions from your previously created message extension app and add it to your own class.